src/test/java/com/github/lwhite1/tablesaw/TableTest.java

```diff
@@ -14,35 +22,61 @@
  */
 public class TableTest {
 
-  private Table table;
-  private FloatColumn column = new FloatColumn("f1");
-
-  @Before
-  public void setUp() throws Exception {
-    table = new Table("t");
-    table.addColumn(column);
-  }
-
-  @Test
-  public void testColumn() throws Exception {
-    Column column1 = table.column(0);
-    assertNotNull(column1);
-  }
-
-  @Test
-  public void testColumnCount() throws Exception {
-    assertEquals(0, new Table("t").columnCount());
-    assertEquals(1, table.columnCount());
-  }
-
-  @Test
-  public void testRowCount() throws Exception {
-    assertEquals(0, table.rowCount());
-    FloatColumn floatColumn = column;
-    floatColumn.add(2f);
-    assertEquals(1, table.rowCount());
-
-    floatColumn.add(2.2342f);
-    assertEquals(2, table.rowCount());
-  }
+    private Table table;
+    private FloatColumn column = new FloatColumn("f1");
+
+    @Before
+    public void setUp() throws Exception {
+        table = new Table("t");
+        table.addColumn(column);
+    }
+
+    @Test
+    public void testColumn() throws Exception {
+        Column column1 = table.column(0);
+        assertNotNull(column1);
+    }
+
+    @Test
+    public void testColumnCount() throws Exception {
+        assertEquals(0, new Table("t").columnCount());
+        assertEquals(1, table.columnCount());
+    }
+
+    @Test
+    public void testRowCount() throws Exception {
+        assertEquals(0, table.rowCount());
+        FloatColumn floatColumn = column;
+        floatColumn.add(2f);
+        assertEquals(1, table.rowCount());
+
+        floatColumn.add(2.2342f);
+        assertEquals(2, table.rowCount());
+    }
+
+    @Test
+    public void testGetRow() throws Exception {
+
+        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
+        Path source = TestData.SIMPLE_UNSORTED_DATA.getSource();
+        List<String[]> rawData;
+        try (CSVReader csvReader = new CSVReader(Files.newBufferedReader(source, Charset.defaultCharset()))) {
+            rawData = csvReader.readAll();
+        }
+        // remove the first row from the raw data since tables don't keep column header row as part of their data
+        rawData.remove(0);
+        int maxRows = table.rowCount();
+        for (int rowIndex = 0; rowIndex < maxRows; rowIndex++) {
+            String row = table.getRow(rowIndex);
+            String expectedRow = getCSVfromListOfStrings(rawData.get(rowIndex));
+            assertEquals("Row " + rowIndex + " matches", expectedRow,row);
+        }
```

Comment:
In the test I had to make sure the getRow method worked. So I had to use the direct OpenCVS reader to get the "known" good data. (I could have gotten that data any number of ways, including just hard coding it, but this way seemed OK.)

Then I compare it to the expected data which TableSaw provides. 

Does this make sense?